### ISSUE TITLE
Some videos are played as pcm 2.0 with Passthrough

### ISSUE THREAD

**Issue by fredbcode** [nova-video-player/aos-AVP#406]
Hello,

With Android TV Philips (the one) I'm using internal player without any issue.
But with nova player with or without Passthrough mode 2, some videos (AAC 5.1) are played only as pcm 2.0

There is no information at all in the integrated player but my av shows dolby digital + 5.1 (with internal player)

The situation is worst with vlc all the videos are played as 2.0, Passthrough or not

With nova many videos are working well with 5.1, but not all

Thanks

**Comment by LostSoul99** [comment]
I am having the exact same issue except every video I have tried that are AAC 5.1 are played as PCM 2.0.  I am using the ARC to send the sound to my Onkyo receiver.  I have tried all 3 audio modes in your app.  If I use the built in media player app they play as DD 5.1.

I am using your app on a Sony XBR-930E TV updated with the latest firmware.

Another issue I am having with my 4k video files encoded with AAC 5.1 is your app will not play them over my network using UpnP.  All other video files work fine.  In the file info section it will say "Did not manage to access the file".  The built in media player app can stream them just fine.

I hope you can fix these issues otherwise I think your app is great since it is so fast.

**Comment by fredbcode** [comment]
Seems similar, do you have also the same issue with vlc ?

**Comment by LostSoul99** [comment]
Exact same issue with VLC, the only media app on an android TV that can output sound in 5.1 is the default one for me

**Comment by courville** [comment]
You are using ARC to connect your TV to your external AV receiver for multichannel I guess. In this case, note that AAC5.1 is likely not supported resulting in PCM 2.0.
I suspect that your native player does re-encoding to AC3 in order to get 5.1 through.
Perhaps with eARC it will be better.
Note that when not using and AndroidTV TV but an AndroidTV box, you could get AAC 5.1 converted to multichannel PCM (not using passthrough) through HDMI (not using ARC) but with the box connected directly to your AV receiver. This is the most stable mode for now (but does not apply in your particular case) cf. https://github.com/nova-video-player/aos-AVP/wiki/Audio-and-passthrough

**Comment by fredbcode** [comment]
I will check (I'm not sure) but I think with spdif output the result still the same, I tested before but maybe something was wrong

**Comment by fredbcode** [comment]
You are right, ARC is a very limited method 

I made a test with spdif all works well

**Comment by courville** [comment]
@fredbcode one question: for sdpif are you using nova passthrough option or not?

**Comment by fredbcode** [comment]
Sorry, I'm wrong, AAC videos still in PCM 2.0 mode, with ARC all the videos are, now with spdif AC3 works well  
I'm using passtrough mode 2

**Comment by fredbcode** [comment]
And the native player still working with Dolby Digital 5.1 with SPDIF

**Comment by courville** [comment]
It means that the native player is re-encoding in ac3 on the fly.
This is a nice feature enabling to go around the limitations of ARC with respect to bandwidth and codecs supported by your AV receiver.

**Comment by fredbcode** [comment]
Yes nice feature, but I disabled ARC, and AAC is supported by my AV Yamaha RX-V479 

The sound seems better in Amazon videos, not yet tested with Netflix. I think my AV displayed "Dolby Digital +" and now "Dolby Digital" (I just tested one movie so I'm not sure)
